### PR TITLE
Add boefje for ADR-validator

### PR DESCRIPTION
**Pull request by hugoboos**
This will add a new boefje that enables users to validate their API's against the Dutch API Design Rules.

It is a work in progress as it needs the correct OOI's.

Model changes in: https://github.com/minvws/nl-kat-octopoes/pull/96

### CODE REVIEW

**Comment by hugoboos** [comment]
The current version of the ADR-validator exits with a 1 when not all rules pass. There is a change in the works to change that behavior. That is why the wip commit is there.

File: tests/test_adr_validator.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+        self.assertEqual(1, len(output.observations))
+
+        observation = output.observations[0]
+        self.assertEqual(8, len(observation.results))
+
+        self.assertEqual("APIDesignRule", observation.results[2].object_type)
+        self.assertEqual("TEST-02", observation.results[2].name)
+        self.assertEqual("APIDesignRuleResult", observation.results[3].object_type)
+        self.assertEqual("An error", observation.results[3].message)
```
Comment (ammar92) [inline tests/test_adr_validator.py:51]: If these aren't strict comparisons, then please use `assertIn` instead

File: tests/test_adr_validator.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        self.assertEqual(False, observation.results[3].passed)
```
Comment (ammar92) [inline tests/test_adr_validator.py:52]:
```suggestion
        self.assertFalse(observation.results[3].passed)
```

File: tests/test_adr_validator.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+        self.assertEqual("ADRFindingType", observation.results[4].object_type)
+        self.assertEqual("TEST-02", observation.results[4].id)
+        self.assertEqual("Finding", observation.results[5].object_type)
+        self.assertEqual("An error", observation.results[5].description)
+        self.assertEqual("ADRFindingType|TEST-02", str(observation.results[5].finding_type))
```
Comment (ammar92) [inline tests/test_adr_validator.py:58]: See comment above

**Review by ammar92 — DISMISSED** [review]
Looks good! Just a few remarks to consider for the tests, but consider it approved

File: tests/test_adr_validator.py (hunk 1 above)
Comment (hugoboos) [reply tests/test_adr_validator.py:51]: These are. I'm testing if the message from the validator is correctly set to the object and finding.

File: tests/test_adr_validator.py (hunk 2 above)
Comment (hugoboos) [reply tests/test_adr_validator.py]: Fixed

**Review by ammar92 — APPROVED** [review]
No more remarks!

**Review by underdarknl — APPROVED** [review]

**Pull request merged**